service/tbc/tbc.go

```diff
@@ -1166,46 +1325,42 @@ func (s *Server) handleHeaders(ctx context.Context, p *peer, msg *wire.MsgHeader
 		}
 
 		// Ask for more block headers at the fork tip.
-		if err = s.getHeaders(ctx, p, lbh.Header); err != nil {
-			log.Errorf("get headers fork: %v", err)
-			return
+		if err = s.getHeaders(ctx, p, lbh.BlockHash()); err != nil {
+			return fmt.Errorf("get headers fork: %v", err)
 		}
 
 		// Also ask for more block headers at canonical tip
-		if err = s.getHeaders(ctx, p, cbh.Header); err != nil {
-			log.Errorf("get headers canonical: %v", err)
-			return
+		if err = s.getHeaders(ctx, p, cbh.BlockHash()); err != nil {
+			return fmt.Errorf("get headers canonical: %v", err)
 		}
 
 	default:
-		// XXX can't happen
-		log.Errorf("invalid insert type: %d", it)
-		return
+		// Can't happen.
+		return fmt.Errorf("invalid insert type: %d", it)
 	}
 
-	// XXX we probably don't want top print it
-	log.Infof("Inserted (%v) %v block headers height %v",
-		it, len(headers), height)
+	log.Infof("Inserted (%v) %v block headers height %v", it, n, height)
+
+	return nil
 }
 
-func (s *Server) handleBlock(ctx context.Context, p *peer, msg *wire.MsgBlock) {
+func (s *Server) handleBlock(ctx context.Context, p *peer, msg *wire.MsgBlock, raw []byte) error {
 	log.Tracef("handleBlock (%v)", p)
 	defer log.Tracef("handleBlock exit (%v)", p)
 
 	block := btcutil.NewBlock(msg)
 	bhs := block.Hash().String()
 	rawBlock, err := block.Bytes()
 	if err != nil {
-		log.Errorf("Unable to get raw block %v: %v", bhs, err)
-		return
+		return fmt.Errorf("handle block unable to get raw block %v: %w", bhs, err)
 	}
 
 	if s.cfg.BlockSanity {
 		err := blockchain.CheckBlockSanity(block, s.chainParams.PowLimit,
 			s.timeSource)
 		if err != nil {
-			log.Errorf("Unable to validate block hash %v: %v", bhs, err)
-			return
+			return fmt.Errorf("handle block unable to validate block hash %v: %v",
```

Comment:
I think the `%w` verb can be used here to wrap the error:
```suggestion
			return fmt.Errorf("handle block unable to validate block hash %v: %w",
```